src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,120 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_SIZE = 16;
+    private static final int RESIZE_CONSTANT = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int threshold;
+    private int size;
+    private Node<K, V>[] arrayNode;
+
+    public MyHashMap() {
+        arrayNode = new Node[DEFAULT_SIZE];
+        threshold = (int) (DEFAULT_SIZE * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            resize();
+        }
+        System.out.println(arrayNode.length + " " + key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        putElement(key, value);
     }
 
     @Override
     public V getValue(K key) {
+        int index = (key == null) ? 0 : getIndex(key);
+        Node<K, V> current = arrayNode[index];
+        while (current != null) {
+            if (Objects.equals(current.key, key)) {
+                return current.value;
+            }
+            current = current.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode()) % arrayNode.length;
+    }
+
+    private void putElement(K key, V value) {
+        Node<K, V> newNode = new Node<>(key, value, null);
+        int index = (key == null) ? 0 : getIndex(key);
+
+        if (arrayNode[index] == null) {
+            arrayNode[index] = newNode;
+            size++;
+        } else {
+            Node<K, V> current = arrayNode[index];
+            while (current.next != null && !Objects.equals(current.key, key)) {
+                current = current.next;
+            }
+            if (Objects.equals(current.key, key)) {
+                current.value = value;
+            } else {
+                current.next = newNode;
+                size++;
+            }
+        }
```

Comment:
nested code is harder to follow. Consider some refactoring, like returning in the first clause and thus eliminating the need for `else`